controllers/managedmcg_controller.go

```diff
@@ -552,3 +568,38 @@ func (r *ManagedMCGReconciler) getCSVByPrefix(name string) (*opv1a1.ClusterServi
 	}
 	return csv, nil
 }
+
+func (r *ManagedMCGReconciler) ensureConsolePlugin(clusterVersion string) error {
+	// The base path to where the request are sent
+	basePath := console.GetBasePath(clusterVersion)
+
+	// Get mcg console Deployment
+	mcgConsoleDeployment := console.GetDeployment(OperatorNamespace)
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      mcgConsoleDeployment.Name,
+		Namespace: mcgConsoleDeployment.Namespace,
+	}, mcgConsoleDeployment)
+	if err != nil {
+		return err
+	}
+
+	// Create/Update mcg console Service
+	mcgConsoleService := console.GetService(r.ConsolePort, OperatorNamespace)
+	_, err = controllerutil.CreateOrUpdate(context.TODO(), r.Client, mcgConsoleService, func() error {
+		return controllerutil.SetControllerReference(mcgConsoleDeployment, mcgConsoleService, r.Scheme)
+	})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	// Create/Update mcg console ConsolePlugin
+	mcgConsolePlugin := console.GetConsolePluginCR(r.ConsolePort, basePath, OperatorNamespace)
+	_, err = controllerutil.CreateOrUpdate(context.TODO(), r.Client, mcgConsolePlugin, func() error {
+		return nil
+	})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	return nil
+}
```

Comment:
who is creating these resources ??
shouldn't this be done when operator is installed, we add deployment to operator CSV for that in case of ODF operator.

Reply:
does here we are doing it via `config/console/kustomization.yaml` file ?

Reply:
@SanjalKatiyar you mean mcg console pod ? That is part of `bundle/manifests/mcg-osd-deployer.clusterserviceversion.yaml` as `deployments`